library/patient.inc

```diff
@@ -1819,6 +1822,31 @@ function getEffectiveInsurances($patient_id, $encdate) {
   return $insarr;
 }
 
+/**
+ * Get all insurance companies
+ *
+ *
+ */
+
+function getAllinsurances($pid){
+   $insarr = array();
+   $sql = "SELECT a.type, a.provider, a.plan_name, a.policy_number, a.group_number,
+           a.subscriber_lname, a.subscriber_fname, a.subscriber_relationship, a.subscriber_employer,
+           b.name, c.line1, c.line2, c.city, c.state, c.zip
+           FROM `insurance_data` AS a
+           RIGHT JOIN insurance_companies AS b
+           ON a.provider = b.id
+           RIGHT JOIN addresses AS c
+           ON a.provider = c.foreign_id
```

Comment:
You could also grab (this may be useful) the default x12 partner of the insurance company.